tests/helper.py

```diff
@@ -90,18 +90,29 @@ def assert_failed(self, with_errors=False, with_warnings=None):
             fail_on_warnings=with_warnings or with_warnings is None), \
                 "Test did not fail; failure was expected."
 
-        if with_errors:
-            assert self.err.errors, "Errors were expected."
-        elif self.err.errors:
-            raise AssertionError("Tests found unexpected errors: %s" %
-                                self.err.print_summary(verbose=True))
-
-        if with_warnings is not None:
-            if with_warnings:
-                assert self.err.warnings, "Warnings were expected."
-            elif self.err.warnings:
-                raise ("Tests found unexpected warnings: %s" %
-                           self.err.print_summary())
+        def check_message(message, props):
+            return all(message.get(prop) == value
+                       for prop, value in props.iteritems())
+
+        def find_message(messages, props):
+            return any(check_message(message, props)
+                       for message in messages)
+
+        def test_messages(mtype, expected):
+            messages = getattr(self.err, mtype)
+
+            if isinstance(expected, (list, tuple)):
```

Comment:
On May 5, 2015 6:47 AM, "Mathieu Agopian" <notifications@github.com> wrote:
>
> In tests/helper.py:
>
> > -                assert self.err.warnings, "Warnings were expected."
> > -            elif self.err.warnings:
> > -                raise ("Tests found unexpected warnings: %s" %
> > -                           self.err.print_summary())
> > +        def check_message(message, props):
> > +            return all(message.get(prop) == value
> > +                       for prop, value in props.iteritems())
> > +
> > +        def find_message(messages, props):
> > +            return any(check_message(message, props)
> > +                       for message in messages)
> > +
> > +        def test_messages(mtype, expected):
> > +            messages = getattr(self.err, mtype)
> > +
> > +            if isinstance(expected, (list, tuple)):
>
> This could be good enough for test code, but the proper way is to do
something like http://stackoverflow.com/a/1952655

I'd already changed this locally :)